frontend/src/components/modalContent/EditVarianceModal.js

```diff
@@ -1,32 +1,95 @@
-import React from "react";
+import React, { Component } from "react";
 import PropTypes from "prop-types";
+import { connect } from "react-redux";
+import { bindActionCreators } from "redux";
 import CustomPropTypes from "@/customPropTypes";
 import EditVarianceForm from "@/components/Forms/variances/EditVarianceForm";
+import {
+  fetchVariancesById,
+  removeDocumentFromVariance,
+  fetchVariancesByMine,
+} from "@/actionCreators/varianceActionCreator";
+import { getVariance } from "@/selectors/varianceSelectors";
 
 const propTypes = {
   onSubmit: PropTypes.func.isRequired,
   closeModal: PropTypes.func.isRequired,
   mineGuid: PropTypes.string.isRequired,
   mineName: PropTypes.string.isRequired,
   coreUsers: CustomPropTypes.options.isRequired,
-  variance: CustomPropTypes.variance.isRequired,
+  variance: CustomPropTypes.variance,
   varianceStatusOptions: CustomPropTypes.options.isRequired,
-  initialValues: CustomPropTypes.variance.isRequired,
+  fetchVariancesById: PropTypes.func.isRequired,
+  varianceGuid: PropTypes.string.isRequired,
+  removeDocumentFromVariance: PropTypes.func.isRequired,
+  fetchVariancesByMine: PropTypes.func.isRequired,
 };
 
-export const EditVarianceModal = (props) => (
-  <EditVarianceForm
-    onSubmit={props.onSubmit}
-    closeModal={props.closeModal}
-    mineGuid={props.mineGuid}
-    mineName={props.mineName}
-    coreUsers={props.coreUsers}
-    variance={props.variance}
-    varianceStatusOptions={props.varianceStatusOptions}
-    initialValues={props.initialValues}
-  />
-);
+const defaultProps = {
+  variance: {},
+};
+
+export class EditVarianceModal extends Component {
+  state = { isLoaded: false };
+
+  componentDidMount() {
+    this.props
+      .fetchVariancesById(this.props.mineGuid, this.props.varianceGuid)
+      .then(this.setState({ isLoaded: true }));
```

Comment:
do we need to put catch statements into these chains

Reply:
no, we're handling the error in the `actionCreator`, I will only set the state if successful, but in saying that, I should add a loading view to the modal to handle this better

Reply:
ended up removing the `.then()` it wasn't doing much without needing a new loading view (which should be implemented site wide anyways)